SchoolManagerSystem.Repository/Implementation/CourseRepository.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SchoolManagerSystem.Common.DTOs;
+using SchoolManagerSystem.Data;
+using SchoolManagerSystem.Model.Entities;
+using SchoolManagerSystem.Repository.Interfaces;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SchoolManagerSystem.Repository.Implementation
+{
+	public class CourseRepository : GenericRepository<Course>, ICourseRepository
+	{
+		private readonly DbSet<Course> _dbSet;
+		public CourseRepository(SMSContext context) : base(context)
+		{
+			_dbSet = context.Set<Course>();
+		}
+
+		public async Task<Course> AddCourse(CourseRequest request, ICollection<Student> students)
+		{
+			var course = new Course
+			{
+				CourseName = request.CourseName,
+				LevelId = request.LevelId,
+				Students = students
```

Comment:
how can the admin add students when creating a course?
just create the course...thats all